src/engine/Server.h

```diff
@@ -172,6 +172,11 @@ class Server {
       SharedCancellationHandle handle, TimeLimit timeLimit,
       const ad_utility::Timer& requestTimer);
 
+  std::pair<std::shared_ptr<ad_utility::websocket::QueryHub>,
+            ad_utility::websocket::MessageSender>
+  createMessageSender(auto& queryHub_, const auto& request,
+                      const string& operation);
+
```

Comment:
Some comments, and maybe you can explicitly type the `queryHub` (weak_ptr<QueryHub> maybe?) and constrain the concept of the `request`